src/client/dfuse/dfuse_cont.c

```diff
@@ -27,123 +28,79 @@ dfuse_cont_open(fuse_req_t req, struct dfuse_inode_entry *parent,
 
-	D_ALLOC_PTR(dfs);
-	if (!dfs)
-		D_GOTO(err, rc = ENOMEM);
-
-	dfuse_dfs_init(dfs, parent->ie_dfs);
-
-	dfs->dfs_dfp = dfp;
-
 	/* Dentry names where are not valid uuids cannot possibly be added so in
 	 * this case return the negative dentry with a timeout to prevent future
 	 * lookups.
 	 */
-	if (uuid_parse(name, dfs->dfs_cont) < 0) {
+	if (uuid_parse(name, cont) < 0) {
 		struct fuse_entry_param entry = {.entry_timeout = 60};
 
 		DFUSE_TRA_INFO(parent, "Invalid container uuid");
 		DFUSE_REPLY_ENTRY(parent, req, entry);
-		D_FREE(dfs);
 		return;
 	}
 
-	DFUSE_TRA_UP(dfs, fs_handle, "dfs");
-
-	D_MUTEX_LOCK(&fs_handle->dpi_info->di_lock);
+	DFUSE_TRA_DEBUG(parent, "Lookup of "DF_UUID" create %d",
+			DP_UUID(cont), create);
 
 	if (create) {
-		rc = dfs_cont_create(dfp->dfp_poh, dfs->dfs_cont,
-				     NULL, &dfs->dfs_coh, &dfs->dfs_ns);
+		D_ALLOC_PTR(dfc);
+		if (!dfc)
+			D_GOTO(err, rc = ENOMEM);
+
+		rc = dfs_cont_create(dfp->dfp_poh, cont, NULL,
+				     &dfc->dfs_coh, &dfc->dfs_ns);
 		if (rc) {
-			DFUSE_TRA_ERROR(dfs,
+			DFUSE_TRA_ERROR(parent,
 					"dfs_cont_create() failed: (%d)",
 					rc);
-			D_GOTO(err_unlock, rc);
-		}
-		d_list_add(&dfs->dfs_list, &dfp->dfp_dfs_list);
-		D_GOTO(alloc_ie, 0);
-	} else {
-		d_list_for_each_entry(dfsi,
-				      &dfp->dfp_dfs_list,
-				      dfs_list) {
-			{
-				struct fuse_entry_param	entry = {0};
-
-				DFUSE_TRA_DEBUG(parent, "Checking %p", dfsi);
-
-				if (uuid_compare(dfsi->dfs_cont,
-						 dfs->dfs_cont) != 0)
-					continue;
-
-				DFUSE_TRA_INFO(parent,
-					       "Found existing container dfs %p",
-					       dfsi);
-
-				rc = dfuse_check_for_inode(fs_handle, dfsi,
-							   &ie);
-				if (rc == -DER_NONEXIST) {
-					D_FREE(dfs);
-					dfs = dfsi;
-					D_GOTO(alloc_ie, 0);
-				} else {
-					D_ASSERT(rc == -DER_SUCCESS);
-				}
-
-				DFUSE_TRA_INFO(ie,
-					       "Reusing existing container entry without reconnect");
-
-				/* Update the stat information, but copy in the
-				 * inode value afterwards.
-				 */
-				rc = dfs_ostat(ie->ie_dfs->dfs_ns, ie->ie_obj,
-					       &entry.attr);
-				if (rc) {
-					DFUSE_TRA_ERROR(ie,
-							"dfs_ostat() failed: (%s)",
-							strerror(rc));
-					D_GOTO(err_unlock, rc);
-				}
-
-				entry.attr.st_ino = ie->ie_stat.st_ino;
-				entry.generation = 1;
-				entry.ino = entry.attr.st_ino;
-				DFUSE_REPLY_ENTRY(ie, req, entry);
-				D_MUTEX_UNLOCK(&fs_handle->dpi_info->di_lock);
-				D_FREE(dfs);
-				return;
-			}
+			D_GOTO(err, rc);
 		}
 	}
 
-	rc = daos_cont_open(dfp->dfp_poh, dfs->dfs_cont,
-			    DAOS_COO_RW, &dfs->dfs_coh, NULL, NULL);
-	if (rc == -DER_NONEXIST) {
-		DFUSE_TRA_INFO(dfs, "daos_cont_open() failed: (%d)", rc);
-		D_GOTO(err_unlock, rc = daos_der2errno(rc));
-	} else if (rc != -DER_SUCCESS) {
-		DFUSE_TRA_ERROR(dfs, "daos_cont_open() failed: (%d)", rc);
-		D_GOTO(err_unlock, rc = daos_der2errno(rc));
-	}
+	rc = dfuse_cont_open(fs_handle, dfp, &cont, &dfc);
+	if (rc)
```

Comment:
If you push again...
```suggestion
	if (rc != 0)
```